kafka-connect-aws-s3/src/fun/scala/io/lenses/streamreactor/connect/S3Utils.scala

```diff
@@ -4,46 +4,74 @@ import com.jayway.jsonpath.internal.Utils.unescape
 import _root_.io.circe.parser._
 import _root_.io.lenses.streamreactor.connect.model.Order
 import _root_.io.lenses.streamreactor.connect.testcontainers.S3Authentication
+import cats.effect.IO
+import cats.effect.Resource
 import org.apache.commons.io.IOUtils
 import org.scalatest.EitherValues
 import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
 import software.amazon.awssdk.regions.Region
-import software.amazon.awssdk.services.s3.model.{BucketLocationConstraint, CreateBucketConfiguration, CreateBucketRequest, GetObjectRequest}
-import software.amazon.awssdk.services.s3.{S3Client, S3Configuration}
+import software.amazon.awssdk.services.s3.model.BucketLocationConstraint
+import software.amazon.awssdk.services.s3.model.CreateBucketConfiguration
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest
+import software.amazon.awssdk.services.s3.model.CreateBucketResponse
+import software.amazon.awssdk.services.s3.model.DeleteBucketRequest
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.S3Configuration
 
 import java.io.InputStream
 import java.net.URI
+import scala.util.Try
 
 object S3Utils extends EitherValues {
 
-  def createBucket(s3Client: S3Client, bucketName: String) = {
-    s3Client.createBucket(
-      CreateBucketRequest
-        .builder()
-        .bucket(bucketName)
-        .createBucketConfiguration(
-          CreateBucketConfiguration.builder()
-            .locationConstraint(BucketLocationConstraint.EU_WEST_1)
-            .build()
+  def createBucket(s3Client: S3Client, bucketName: String): Resource[IO, CreateBucketResponse] =
+    Resource.make(
+      IO {
+        s3Client.createBucket(
+          CreateBucketRequest
+            .builder()
+            .bucket(bucketName)
+            .createBucketConfiguration(
+              CreateBucketConfiguration.builder()
+                .locationConstraint(BucketLocationConstraint.EU_WEST_1)
+                .build(),
+            )
+            .build(),
         )
-        .build())
-  }
+      },
+    ) { _ =>
+      IO {
```

Comment:
IO.fromTry(Try(...)), otherwise the exception is ignored. It's best to allow the the recoverWith on the IO than hiding the excetion

Reply:
Great suggestion, thank you.